### PR TITLE
Fix custom type producing an error because there was no template option

### PR DESCRIPTION
**Pull request by mosanger**
depends on and includes #42 

Introduced a new config option `templateName` when using `type: custom`
Don't really know if this was the intended usage, but it wasn't working anyway.
Feel free to rename the option to anything else ;)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -78,7 +90,10 @@ async function startTemplateGenerator() {
     }
 
     if (results.type === 'custom') {
-      await generateFilesFromCustom(results)
+      const { templateName } = config
+      if (!templateName) throw new Error('Please add a templateName to the config if using the custom type')
```
Comment (CVarisco) [inline src/index.js]: According to:[docs](https://github.com/CVarisco/create-component-app/blob/master/docs/CUSTOM-TEMPLATES.md) 
What happens if the user doesn't create a subfolder on the "custom" template dir ?
Do you understand what I mean ?

For example:

customTemplateDir
   |---index.js

```json
{
  "type":"custom",
  "template": ?????,
  "templatesDirPath": "./customTemplateDir"
}
```

Thanks a lot for you support 👍

Comment (mosanger) [reply src/index.js]: if the template does not exist, getTemplate will throw an error if that's what you mean

Comment (mosanger) [reply src/index.js]: But I didn't test what happens when the templatesDirPath itself doesn't exist

Comment (CVarisco) [reply src/index.js]: Thanks man !
I'll merge now 🚜

**Comment by CVarisco** [comment]
Ok, I'm tested and works very well!
I'll change the documentation to specify that the user needs to create a subfolder now 👍

File: src/index.js (hunk 1 above)
Comment (mosanger) [reply src/index.js]: oh... so you were planning to have either one template in templatesDirPath, or several?
how would you distinguish? either way, type custom wasn't really working was it? (at least I didn't get it to work without throwing an error)

Comment (mosanger) [reply src/index.js]: and thanks for merging ;)

Comment (CVarisco) [reply src/index.js]: Sure! You are right!
The first implementation was different and maybe all the changes brake something.

However, now it is better :)
Thank you 💻

**Pull request merged**